satpy/readers/hdfeos_mod35.py

```diff
@@ -0,0 +1,434 @@
+import logging
+from datetime import datetime
+
+import numpy as np
+from pyhdf.error import HDF4Error
+from pyhdf.SD import SD, SDC
+
+import dask.array as da
+import xarray.ufuncs as xu
+import xarray as xr
+from satpy import CHUNK_SIZE
+from satpy.dataset import DatasetID
+from satpy.readers.file_handlers import BaseFileHandler
+from satpy.readers.hdf4_utils import from_sds
+
+logger = logging.getLogger(__name__)
+
+import time
+
+class HDFEOSFileReader(BaseFileHandler):
+
+    def __init__(self, filename, filename_info, filetype_info):
+        super(HDFEOSFileReader, self).__init__(filename, filename_info, filetype_info)
+        self.filename = filename
+        try:
+            self.sd = SD(str(self.filename), SDC.READ)
+        except HDF4Error as err:
+            raise ValueError("Could not load data from " + str(self.filename)
+                             + ": " + str(err))
+        self.metadata = self.read_mda(self.sd.attributes()['CoreMetadata.0'])
+        self.metadata.update(self.read_mda(
+            self.sd.attributes()['StructMetadata.0']))
+        self.metadata.update(self.read_mda(
+            self.sd.attributes()['ArchiveMetadata.0']))
+
+    @property
+    def start_time(self):
+        date = (self.metadata['INVENTORYMETADATA']['RANGEDATETIME']['RANGEBEGINNINGDATE']['VALUE'] + ' ' +
```

Comment:
Is this and `end_time` any different from the corresponding hdfeos l1 reader counterpart ? Maybe creating a common base class can be an easy way to factorise some code ?

Reply:
I had the chance to have a look at possible refactorisation. As far as I can see the only difference between the HDFEOSFileReader class here compared to the l1b version is in line 56+57. When I remember right this is the reason I copied the class over because I didn't want to modify an existing reader class (also I don't know if/what effects those two lines would have on the l1b reader). So if those two lines could be included in that class the two readers could use that. 
The reason for the introduction is that the mod35_l2 files have some nasty attributes with line breaks and what not which don't hold any valuable information and are skipped with those lines.

Further more I thought the HDFEOSGeoReader class could be shared. While testing this though (with the updated class using geotiepoints) I discovered that for whatever reason the mod35_l2 files sometimes differ in the dimension size of the arrays compared to the l1b data. So when no geolocation files are supplied and the coordinates should be interpolated the ModisInterpolator interpolation errors because the cscan_full_width is hardwired to 271. I don't know if it makes sense to make that dynamic or for the time being in order to use the mod35 reader geolocation files are a requirement?